docs: add homepage of https://numaflow.numaproj.io/ (#163)
numaproj/numaflow: docs/README.md

> # Numaflow

Author: @edlee2121 - we need this file as it's the homepage of numaproj.github.io/numaflow/.

Please suggest the content.

> Numaflow is a Kubernetes-native platform for running massive parallel data processing and streaming jobs.

Reviewer: ```suggestion
Numaflow is a Kubernetes-native platform for running massively parallel data processing and streaming jobs.
```

> A Numaflow Pipeline is implemented as a Kubernetes custom resource, and consists of one or more source, data processing, and sink vertices.

Reviewer: ```suggestion
A Numaflow Pipeline is implemented as a Kubernetes custom resource and consists of a source vertex, multiple data processing vertices, and sink vertices.
```
